pkg/storage/bucket/client_test.go

```diff
@@ -127,3 +129,99 @@ func TestClientMock_MockGet(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestClient_ConfigValidation(t *testing.T) {
+	testCases := []struct {
+		name           string
+		cfg            Config
+		expectingError bool
+	}{
+		{
+			name:           "valid storage_prefix",
+			cfg:            Config{Backend: Filesystem, StoragePrefix: "helloworld"},
+			expectingError: false,
+		},
+		{
+			name:           "storage_prefix non-alphanumeric characters",
+			cfg:            Config{Backend: Filesystem, StoragePrefix: "hello-world!"},
+			expectingError: true,
+		},
+		{
+			name:           "storage_prefix suffixed with a slash (non-alphanumeric)",
+			cfg:            Config{Backend: Filesystem, StoragePrefix: "helloworld/"},
+			expectingError: true,
+		},
+		{
+			name:           "storage_prefix that has some character strings that have a meaning in unix paths (..)",
+			cfg:            Config{Backend: Filesystem, StoragePrefix: ".."},
+			expectingError: true,
+		},
+		{
+			name:           "storage_prefix that has some character strings that have a meaning in unix paths (.)",
+			cfg:            Config{Backend: Filesystem, StoragePrefix: "."},
+			expectingError: true,
+		},
+		{
+			name:           "unsupported backend",
+			cfg:            Config{Backend: "flash drive"},
+			expectingError: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			if actualErr := tc.cfg.Validate(); tc.expectingError {
+				assert.Error(t, actualErr)
+			} else {
+				assert.NoError(t, actualErr)
+			}
+		})
+	}
+}
+
+func TestNewPrefixedBucketClient(t *testing.T) {
+	t.Run("with prefix", func(t *testing.T) {
+		ctx := context.Background()
+		tempDir := t.TempDir()
+		cfg := Config{
+			Backend:       Filesystem,
+			StoragePrefix: "prefix",
+			Filesystem: filesystem.Config{
+				Directory: tempDir,
+			},
+		}
+
+		client, err := NewClient(ctx, cfg, "test", util_log.Logger, nil)
+		require.NoError(t, err)
+
+		err = client.Upload(ctx, "file", bytes.NewBufferString("content"))
+		assert.NoError(t, err)
+
+		_, err = client.Get(ctx, "file")
+		assert.NoError(t, err)
+
+		assert.FileExists(t, tempDir+"/prefix/file")
+	})
+
+	t.Run("without prefix", func(t *testing.T) {
+		ctx := context.Background()
+		tempDir := t.TempDir()
+		cfg := Config{
+			Backend: Filesystem,
+			Filesystem: filesystem.Config{
+				Directory: tempDir,
+			},
+		}
+
+		client, err := NewClient(ctx, cfg, "test", util_log.Logger, nil)
+		require.NoError(t, err)
+		err = client.Upload(ctx, "file", bytes.NewBufferString("content"))
+		require.NoError(t, err)
+
+		_, err = client.Get(ctx, "file")
+		assert.NoError(t, err)
+
+		assert.FileExists(t, tempDir+"/file")
+	})
```

Comment:
`bucket.NewClient` is only used in these tests + https://github.com/grafana/mimir/blob/b1b9882379b899a317425ed7c4aaa9e308498375/pkg/compactor/split_merge_compactor_test.go#L577 

Other tests may use `NewUserBucketClient` and `NewPrefixedBucketClient`

I think the usage in the compactor test is not intentionally testing whether the prefix access works as intended. It will fail if bucket access is broken, but it won't be obvious why it failed.